python/taichi/lang/ops.py

```diff
@@ -373,7 +374,11 @@ def round(x):  # pylint: disable=redefined-builtin
 
 
 @unary
-def floor(x):
+def _floor(x):
+    return _unary_operation(_ti_core.expr_floor, math.floor, x)
+
+
+def floor(x, dtype: template() = float):
```

Comment:
why is it a template?

Reply:
Oh, I used template because I was implementing it as a `ti.func`. It's fixed now.